pytorch_lightning/trainer/trainer.py

```diff
@@ -2640,7 +2642,30 @@ def terminate_on_nan(self, val: bool) -> None:
         self._terminate_on_nan = val  # : 212
 
 
-def _determine_batch_limits(batches: Union[int, float], name: str) -> Union[int, float]:
+def _determine_batch_limits(batches: Optional[Union[int, float]], name: str) -> Union[int, float]:
+    if batches is None:
+        # batches is optional to know if the user passed a value so that we can show the above info messages only to the
+        # users that set a value explicitly
+        return 1.0
```

Comment:
given these are now optional on the input, does it make sense to later on refactor the loops to only check these if they're set?

Reply:
Yes. IMO these should be moved to the loops that use them.

Reply:
With a similar reasoning to #11845